src/server/project.ts

```diff
@@ -211,52 +214,162 @@ namespace ts.server {
         }
 
-        private watchDirectoryOfFailedLookup(directory: string) {
+        getCompilationSettings() {
+            return this.compilerOptions;
+        }
+
+        getNewLine() {
+            return this.partialSystem.newLine;
+        }
+
+        getProjectVersion() {
+            return this.projectStateVersion.toString();
+        }
+
+        getScriptFileNames() {
+            const result: string[] = [];
+            if (this.rootFiles) {
+                this.rootFilesMap.forEach((value, _path) => {
+                    const f: ScriptInfo = isScriptInfo(value) && value;
+                    if (this.languageServiceEnabled || (f && f.isScriptOpen())) {
+                        // if language service is disabled - process only files that are open
+                        result.push(f ? f.fileName : value as NormalizedPath);
+                    }
+                });
+                if (this.typingFiles) {
+                    for (const f of this.typingFiles) {
+                        result.push(f);
+                    }
+                }
+            }
+            return result;
+        }
+
+        private getScriptInfoLSHost(fileName: string) {
+            const scriptInfo = this.projectService.getOrCreateScriptInfo(fileName, /*openedByClient*/ false, this.partialSystem);
+            if (scriptInfo) {
+                const existingValue = this.rootFilesMap.get(scriptInfo.path);
+                if (existingValue !== undefined && existingValue !== scriptInfo) {
+                    // This was missing path earlier but now the file exists. Update the root
+                    this.rootFiles.push(scriptInfo);
+                    this.rootFilesMap.set(scriptInfo.path, scriptInfo);
+                }
+                scriptInfo.attachToProject(this);
+            }
+            return scriptInfo;
+        }
+
+        getScriptKind(fileName: string) {
+            const info = this.getScriptInfoLSHost(fileName);
+            return info && info.scriptKind;
+        }
+
+        getScriptVersion(filename: string) {
+            const info = this.getScriptInfoLSHost(filename);
+            return info && info.getLatestVersion();
+        }
+
+        getScriptSnapshot(filename: string): IScriptSnapshot {
+            const scriptInfo = this.getScriptInfoLSHost(filename);
+            if (scriptInfo) {
+                return scriptInfo.getSnapshot();
+            }
+        }
+
+        getCancellationToken() {
+            return this.projectService.cancellationToken;
+        }
+
+        getCurrentDirectory(): string {
+            return this.partialSystem.getCurrentDirectory();
+        }
+
+        getDefaultLibFileName() {
+            const nodeModuleBinDir = getDirectoryPath(normalizePath(this.projectService.host.getExecutingFilePath()));
+            return combinePaths(nodeModuleBinDir, getDefaultLibFileName(this.compilerOptions));
+        }
+
+        useCaseSensitiveFileNames() {

```

Comment:
Might want to make this one `get useCaseSensitiveFileNames()`?

Reply:
This is the function LanguageServiceHost needs to implement and moved from LSHost to project as that host was redundant now